components/core/src/package/install.rs

```diff
@@ -231,15 +231,27 @@ impl PackageInstall {
         // In most cases, this does nothing and should only mutate
         // the paths in a windows studio where FS_ROOT_PATH will
         // be the studio root path (ie c:\hab\studios\...)
-        for path in &mut paths {
-            *path = fs::fs_rooted_path(&path, &self.fs_root_path);
-        }
+        let rooted_path = self.root_paths(&mut paths)?;
 
-        let joined = env::join_paths(paths)?.into_string()
-                                            .map_err(Error::InvalidPathString)?;
         // Only insert a PATH entry if the resulting path string is non-empty
-        if !joined.is_empty() {
-            env.insert(PATH_KEY.to_string(), joined);
+        if !rooted_path.is_empty() {
+            env.insert(PATH_KEY.to_string(), rooted_path);
+        }
+
+        // release 0.85.0 introduces the RUNTIME_ENVIRONMENT_PATHS metadata
+        // that are environment variables containing file paths that are to be
+        // rooted under FS_ROOT. For backwards compatibility the key/value
+        // pairs are duplicated in RUNTIME_ENVIRONMENT so we will remove previously
+        // stored values with the same key.
+        let environment_paths =
+            self.runtime_environment_file_map(MetaFile::RuntimeEnvironmentPaths)?;
+        for (key, value) in environment_paths.into_iter() {
+            let mut split_path: Vec<PathBuf> = env::split_paths(&value).collect();
+            let rooted_path = self.root_paths(&mut split_path)?;
+            if !rooted_path.is_empty() {
```

Comment:
Do we ever expect this to be empty, and if it is, is there harm in adding it to `env`? Won't it already be set as empty in `RUNTIME_ENVIRONMENT` so it would already be set as empty in `env`? Just curious and trying to understand the code better.

Reply:
That's a great point. I was basically reproducing the code above that works with `PATH` but was wondering the same thing here. And you bring up a good point that even if it is empty the empty value would have gotten propagated in `RUNTIME_ENVIRONMENT`. I can't think of why it would be empty unless someone either manually manipulated the file or explicitly set the variable to an empty string in `Invoke-SetupEnvironment`. And if they did, they probably did so for a reason. So I'm going to remove this check.